src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeMetadataGenerator.cs

```diff
@@ -87,7 +93,14 @@ public byte[] GenerateEventMetadata(EventMetadata eventMetadata)
                 // Increase the metadataLength for parameters.
                 foreach (var parameter in parameters)
                 {
-                    metadataLength = metadataLength + parameter.GetMetadataLength();
+                    int pMetadataLength = parameter.GetMetadataLength();
```

Comment:
Thanks @brianrob! Quick question -- if we just simply build the default metadata based on what ```pMetadataLength``` is, we may get another failure during the actual metadata generation of parameters. If that happens, do we fall back and re-allocate another blob to generate the default metadata (since the size is now different from what we originally allocated), or should we just set the parameterCount to 0 and let the consumer deal with this mismatch?

Reply:
You should fallback and fix up the length.  Otherwise the consumer won't know how to deal with it.